buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/Utils.kt

```diff
@@ -1,24 +1,115 @@
+ */
+
 package org.jetbrains.kotlin
 
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.jetbrains.kotlin.konan.target.HostManager
 import org.jetbrains.kotlin.konan.target.KonanTarget
 import org.jetbrains.kotlin.konan.target.PlatformManager
+import java.io.File
+
+//region Project properties.
+
+val Project.platformManager
+    get() = findProperty("platformManager") as PlatformManager
+
+val Project.testTarget
+    get() = findProperty("target") as KonanTarget
+
+val Project.verboseTest
+    get() = hasProperty("test_verbose")
 
-fun Project.platformManager() = findProperty("platformManager") as PlatformManager
-fun Project.testTarget() = findProperty("target") as KonanTarget
+val Project.testOutputLocal
+    get() = (findProperty("testOutputLocal") as File).toString()
+
+val Project.testOutputStdlib
+    get() = (findProperty("testOutputStdlib") as File).toString()
+
+val Project.testOutputFramework
+    get() = (findProperty("testOutputFramework") as File).toString()
+
+@Suppress("UNCHECKED_CAST")
+val Project.globalTestArgs: List<String>
+    get() = with(findProperty("globalTestArgs")) {
+            if (this is Array<*>) this.toList() as List<String>
+            else this as List<String>
+    }
+
+//endregion
 
 /**
- * Ad-hoc signing of the specified path
+ * Ad-hoc signing of the specified path.
  */
 fun codesign(project: Project, path: String) {
-    check(HostManager.hostIsMac, { "Apple specific code signing" } )
+    check(HostManager.hostIsMac) { "Apple specific code signing" }
     val (stdOut, stdErr, exitCode) = runProcess(executor = localExecutor(project), executable = "/usr/bin/codesign",
             args = listOf("--verbose", "-s", "-", path))
-    check(exitCode == 0, { """
-            |Codesign failed with exitCode: $exitCode
-            |stdout: $stdOut
-            |stderr: $stdErr
-            """.trimMargin()
-    })
-}
\ No newline at end of file
+    check(exitCode == 0) { """
+        |Codesign failed with exitCode: $exitCode
+        |stdout: $stdOut
+        |stderr: $stdErr
+        """.trimMargin()
+    }
+}
+
+/**
+ * Creates a list of file paths to be compiled from the given [compile] list with regard to [exclude] list.
+ */
+fun Project.getFilesToCompile(compile: List<String>, exclude: List<String>): List<String> {
+    // convert exclude list to paths
+    val excludeFiles = exclude.map { project.file(it).absolutePath }.toList()
+
+    // create list of tests to compile
+    return compile.flatMap { f ->
+        project.file(f)
+                .walk()
+                .filter { it.isFile && it.name.endsWith(".kt") && !excludeFiles.contains(it.absolutePath) }
+                .map(File::getAbsolutePath)
+                .asIterable()
+    }
+}
+
+//region Task dependency.
+
+fun Project.setDistDependencyFor(taskName: String) {
+    project.setDistDependencyFor(project.tasks.getByName(taskName))
+}
+
+fun Project.setDistDependencyFor(t: Task) {
```

Comment:
IMHO making this function an extension for `Task` makes its call more readable: `task.dependsOnDist()` or `task.setDistDependency()`.